src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -1520,7 +1520,7 @@ bool Pet::removeSpell(uint32 spell_id, bool learn_prev, bool clear_ab)
     }
 
     // if remove last rank or non-ranked then update action bar at server and client if need
-    if (m_charmInfo->RemoveSpellFromActionBar(spell_id) && !learn_prev && clear_ab)
```

Comment:
I want the function call to only happen when both ``clear_ab == true`` and  ``!learn_pre == true`` (``learn_prev == false``) or rather I want the order to be restored to what it was before commit 6c21ddd9b98d372cb802c9abf034e2aa3ecc70b9. My patch simply reverts one line of this commit and thereby fixes an issue obviously caused by it, i.e it's 100% reproducable/fixed without/with my patch.
I would be happier myself, if I could give a better explanation, but I just started a few days ago to familiarize myself with the _TrinityCore_ source, actually with the intention to fix this (and maybe some other) issue(s).   .